src/components/AllToDos.tsx

```diff
@@ -0,0 +1,23 @@
+import SingleToDo from "./SingleToDo";
+
+type propsType= {
+    data:{
+        todos:{id: number, title: string; completed: boolean }[],
+        isLoading: boolean,
+        error: string
+     },
+     onCheckClick(id: number): void;
+     onDeleteClick(id: number): void;
+}
+
+
+
+function AllToDos(props: propsType){
+    return (
+        !props.data.isLoading ? <div className="spinner"></div>
+        : <div className="toDos">
+            {props.data.todos.map(todo=> <SingleToDo key={todo.id} todo={todo} onCheckClick={props.onCheckClick} onDeleteClick={props.onDeleteClick}/>)}
```

Comment:
you can do the following instead of passing the whole todo just to pass the id to the handlers

```suggestion
            {props.data.todos.map(todo=> <SingleToDo key={todo.id} todo={todo} onCheckClick={()=>props.onCheckClick(todo.id)} onDeleteClick={()=>props.onDeleteClick(todo.id)}/>)}
```
and make sure to exclude these as explicit handlers